controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go

```diff
@@ -158,6 +181,40 @@ type RollingUpdate struct {
 	MaxSurge *intstr.IntOrString `json:"maxSurge,omitempty"`
 }
 
+// RemediationStrategy allows to define how control plane machines remediation happens.
+type RemediationStrategy struct {
+	// MaxRetry is the Max number of retry while attempting to remediate an unhealthy machine.
+	// A retry happens when a machine that was created as a replacement for an unhealthy machine also fails.
+	// For example, given a control plane with three machines M1, M2, M3:
+	//
+	//	M1 become unhealthy; remediation happens, and M1bis is created as a replacement.
```

Comment:
```suggestion
	//	M1 become unhealthy; remediation happens, and M1-1 is is created as a replacement.
```
I assume (based on the next line)